Flaky Test when seed not used

The test `test_max_truncation_error` in `tensornetwork/backends/symmetric/decompositions_test.py` [link](https://github.com/google/TensorNetwork/blob/edf3f345640c64595b4ad1002bebf4a008988b3e/tensornetwork/backends/symmetric/decompositions_test.py#L90) almost always fails when the seed [here](https://github.com/google/TensorNetwork/blob/edf3f345640c64595b4ad1002bebf4a008988b3e/tensornetwork/backends/symmetric/decompositions_test.py#L91) is removed. This happens only for `[3-float64]` and `[4-complex128]` parameters. The test failed 500/500 times for me for both these parameters.

From the error message it seems `svals[mask][::-1]` is always empty whereas `S2.data` most often has an element of size 1(except when using the seed). Is this expected? Or is this indicative of a bug?

Please let me know if you need more details. I will be happy to raise a PR if you have any suggestions on how to fix it.

Test failure message:
```
>     np.testing.assert_allclose(S2.data, svals[mask][::-1])
E     AssertionError:
E     Not equal to tolerance rtol=1e-07, atol=0
E
E     (shapes (1,), (0,) mismatch)
E      x: array([1.])
E      y: array([], dtype=float64)

tensornetwork/backends/symmetric/decompositions_test.py:112: AssertionError
```

Environment:
```
Anaconda python 3.6
tensorflow==2.6.0
numpy==1.19.5
scipy==1.5.4
```

--- a/tensornetwork/backends/symmetric/decompositions.py
+++ b/tensornetwork/backends/symmetric/decompositions.py
@@ -35,6 +35,7 @@ def svd(
   Computes the singular value decomposition (SVD) of a tensor.
   See tensornetwork.backends.tensorflow.decompositions for details.
   """
+
   left_dims = tensor.shape[:pivot_axis]
   right_dims = tensor.shape[pivot_axis:]
 
@@ -91,6 +92,7 @@ def svd(
       if relative and (len(singvals) > 0):
         max_truncation_error = max_truncation_error * np.max(
             [s[0] for s in singvals])
+
       kept_inds_mask = np.sqrt(
           np.cumsum(np.square(
               extended_flat_singvals[inds]))) > max_truncation_error

--- a/tensornetwork/backends/symmetric/decompositions.py
+++ b/tensornetwork/backends/symmetric/decompositions.py
@@ -84,10 +84,6 @@ def svd(
     #sort singular values
     inds = np.argsort(extended_flat_singvals, kind='stable')
     discarded_inds = np.zeros(0, dtype=SIZE_T)
-    if inds.shape[0] > 0:
-      maxind = inds[-1]
-    else:
-      maxind = 0
     if max_truncation_error is not None:
       if relative and (len(singvals) > 0):
         max_truncation_error = max_truncation_error * np.max(

--- a/tensornetwork/backends/symmetric/decompositions.py
+++ b/tensornetwork/backends/symmetric/decompositions.py
@@ -35,7 +35,6 @@ def svd(
   Computes the singular value decomposition (SVD) of a tensor.
   See tensornetwork.backends.tensorflow.decompositions for details.
   """
-
   left_dims = tensor.shape[:pivot_axis]
   right_dims = tensor.shape[pivot_axis:]
 
@@ -92,7 +91,6 @@ def svd(
       if relative and (len(singvals) > 0):
         max_truncation_error = max_truncation_error * np.max(
             [s[0] for s in singvals])
-
       kept_inds_mask = np.sqrt(
           np.cumsum(np.square(
               extended_flat_singvals[inds]))) > max_truncation_error
@@ -114,9 +112,7 @@ def svd(
 
     if len(inds) == 0:
       #special case of truncation to 0 dimension;
-      warnings.warn("svd_decomposition truncated to 0 dimensions. "
-                    "Adjusting to `max_singular_values = 1`")
-      inds = np.asarray([maxind])
+      warnings.warn("svd_decomposition truncated to 0 dimensions.")
 
     if extended_singvals.shape[1] > 0:
       #pylint: disable=no-member

--- a/tensornetwork/backends/symmetric/decompositions_test.py
+++ b/tensornetwork/backends/symmetric/decompositions_test.py
@@ -87,8 +87,9 @@ def test_max_singular_values(dtype, R, R1, num_charges):
 
 @pytest.mark.parametrize("dtype", np_dtypes)
 @pytest.mark.parametrize("num_charges", [1, 2, 3])
-def test_max_truncation_error(dtype, num_charges):
-  np.random.seed(10)
+@pytest.mark.parametrize("seed", np.arange(20))
+def test_max_truncation_error(dtype, num_charges, seed):
+  np.random.seed(seed)
   R = 2
   D = 30
   charges = [

